profiler/CMakeLists.txt

```diff
@@ -52,6 +52,11 @@ set (srcs
 esma_add_library (${this} SRCS ${srcs} DEPENDENCIES GFTL_SHARED::gftl-shared GFTL::gftl-v1 GFTL::gftl-v2 MAPL.shared MPI::MPI_Fortran TYPE ${MAPL_LIBRARY_TYPE})
 target_include_directories (${this} PRIVATE ${MAPL_SOURCE_DIR}/include)
 
+#if (NOT CMAKE_Fortran_COMPILER_ID MATCHES "NAG")
```

Comment:
Should this line (and others like it) be deleted?

Reply:
I think NAG still has issues with OpenMP and CMake: https://gitlab.kitware.com/cmake/cmake/-/issues/21280

Though I haven't tested in a while.

One question I have is: Where is `OMP` set? I don't see any `option()` or anything for it. And it should probably have a better name as we have code in GEOS that we *always* compile with OpenMP (like, `GEOSgcm.x`), and so this might need a more descriptive name.

Reply:
OMP is set in respective CMakeLists.txt files. Some of the files have if(OMP)/endif() constructs thus OpenMP is enable in those cases by adding -DOMP option to the cmake step. We may enable OpenMP by default in those cases.

Reply:
@aoloso I guess that is good for now, as it is a bit of a "hidden" feature, but if this ever goes to production we should have a much more explicit name for an option and it should be an actual CMake `option()`.

Reply:
@tclune : GFortran error message is

    _ASSERT_RC(this%stack%size()== 1,"Stack not empty when timer stopped.",INCORRECTLY_NESTED_METERS)
      |                                                                                                                                                                                                                       ^
Error: invalid branch to/from OpenMP structured block